auth_mellon_handler.c

```diff
@@ -2761,6 +2761,39 @@ static int am_init_authn_request_common(request_rec *r,
 
     LASSO_PROFILE(login)->msg_relayState = g_strdup(return_to_url);
 
+#ifdef HAVE_ECP
+    {
+        am_req_cfg_rec *req_cfg;
+        req_cfg = am_get_req_cfg(r);
+
+        /*
+         * The signature hint must be set prior to calling
+         * lasso_login_build_authn_request_msg
+         */
+
+        if (req_cfg->ecp_service_options &
+            (ECP_SERVICE_OPTION_WANT_AUTHN_SIGNED |
+             ECP_SERVICE_OPTION_CHANNEL_BINDING)) {
+            /*
+             * authnRequest should be signed if the client requested it
+             * or if channel bindings are enabled.
+             */
+            lasso_profile_set_signature_hint(LASSO_PROFILE(login),
+                                             LASSO_PROFILE_SIGNATURE_HINT_FORCE);
+        } else {
+            /*
+             * The client did not explicitly the request be signed. To
+             * be compatible with earlier versions of ECP which did
+             * not have this flag default to respecting the metadata
+             * and/or the configuration, the lasso
+             * LASSO_PROFILE_SIGNATURE_HINT_MAYBE covers this case.
+             */
+            lasso_profile_set_signature_hint(LASSO_PROFILE(login),
+                                             LASSO_PROFILE_SIGNATURE_HINT_MAYBE);
```

Comment:
This is logically equivalent to unconditionally setting it to LASSO_PROFILE_SIGNATURE_HINT_MAYBE for the ECP case. The default in Lasso is LASSO_PROFILE_SIGNATURE_HINT_MAYBE, as a consequence I've removed this block, the default should be in effect.